src/hb-aat-layout-trak-table.hh

```diff
@@ -64,21 +64,71 @@ struct TrackData
     return_trace (c->check_struct (this));
   }
 
-  protected:
+  public:
   HBUINT16		nTracks;	/* Number of separate tracks included in this table. */
   HBUINT16		nSizes;		/* Number of point sizes included in this table. */
   LOffsetTo<UnsizedArrayOf<Fixed> >
 			sizeTable;
-  TrackTableEntry	trackTable[VAR];/* Array[nSizes] of size values. */
+  UnsizedArrayOf<TrackTableEntry>	trackTable;/* Array[nSizes] of size values. */
 
   public:
   DEFINE_SIZE_ARRAY (8, trackTable);
 };
 
+#define HB_CORETEXT_DEFAULT_FONT_SIZE 12.f
+
+static float
+coretext_font_size_from_ptem (float ptem)
+{
+  /* CoreText points are CSS pixels (96 per inch),
+   * NOT typographic points (72 per inch).
+   *
+   * https://developer.apple.com/library/content/documentation/GraphicsAnimation/Conceptual/HighResolutionOSX/Explained/Explained.html
+   */
+  ptem *= 96.f / 72.f;
+  return ptem <= 0.f ? HB_CORETEXT_DEFAULT_FONT_SIZE : ptem;
+}
+
+
 struct trak
 {
   static const hb_tag_t tableTag = HB_AAT_TAG_trak;
 
+  inline bool apply (hb_aat_apply_context_t *c) const
+  {
+    TRACE_APPLY (this);
+
+    hb_unicode_funcs_t *unicode = c->buffer->unicode;
+    unsigned int u = c->buffer->info->codepoint;
+    unsigned int gen_cat = (unsigned int) unicode->general_category (u);
+    printf(">>> %d %d <<<", u, gen_cat);
+
+    const TrackData trackData = this+horizOffset;
+    const TrackTableEntry trackTableEntry = trackData.trackTable[0];
+
+    for (unsigned int i = 0; i < trackData.nSizes; ++i)
```

Comment:
This fetching of trak should be methods on the respective structs.

Also, you need a track number, which defaults to 0. We should add API on the buffer or shape later to set traking / letter-spacing.

Reply:
Oh, sure

Reply:
> Also, you need a track number, which defaults to 0. We should add API on the buffer or shape later to set traking / letter-spacing.

That would be nice also.